tests/installation/grub_test.pm

```diff
@@ -21,73 +21,10 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
-use utils;
-use version_utils qw(is_sle is_livecd);
-use bootloader_setup qw(stop_grub_timeout boot_into_snapshot);
-
-=head2 handle_installer_medium_bootup
-
-Due to pre-installation setup, qemu boot order is always booting from CD-ROM.
-=cut
-sub handle_installer_medium_bootup {
-    my ($self) = @_;
-
-    return unless (check_var("BOOTFROM", "d") || (get_var('UEFI') && get_var('USBBOOT')));
-    assert_screen 'inst-bootmenu', 180;
-
-    if (check_var("BOOTFROM", "d") && check_var("AUTOUPGRADE") && check_var("PATCH")) {
-        assert_screen 'grub2';
-    }
-
-    # Layout of live is different from installation media
-    my $key = is_livecd() ? 'down' : 'up';
-    send_key_until_needlematch 'inst-bootmenu-boot-harddisk', $key;
-    send_key 'ret';
-
-    # use firmware boot manager of aarch64 to boot upgraded system
-    $self->handle_uefi_boot_disk_workaround if (check_var('ARCH', 'aarch64'));
-}
-
-sub bug_workaround_bsc1005313 {
-    record_soft_failure "Running with plymouth:debug to catch bsc#1005313" if get_var('PLYMOUTH_DEBUG');
-    send_key 'e';
-    # Move to end of kernel boot parameters line
-    send_key_until_needlematch "linux-line-selected", "down";
-    send_key "end";
-
-    assert_screen "linux-line-matched";
-    if (get_var('PLYMOUTH_DEBUG')) {
-        # remove "splash=silent quiet showopts"
-        for (1 .. 28) { send_key "backspace" }
-        type_string 'plymouth:debug';
-    }
-    type_string " " . get_var('GRUB_KERNEL_OPTION_APPEND') if get_var('GRUB_KERNEL_OPTION_APPEND');
-
-    save_screenshot;
-    send_key 'ctrl-x';
-}
+use grub_utils qw(grub_test);
 
 sub run {
-    my ($self) = @_;
-    my $timeout = get_var('GRUB_TIMEOUT', 90);
-
-    $self->handle_installer_medium_bootup;
-    workaround_type_encrypted_passphrase;
-    # 60 due to rare slowness e.g. multipath poo#11908
-    # 90 as a workaround due to the qemu backend fallout
-    assert_screen_with_soft_timeout('grub2', timeout => 2 * $timeout, soft_timeout => $timeout, bugref => 'boo#1120256');
-    stop_grub_timeout;
-    boot_into_snapshot                                               if get_var("BOOT_TO_SNAPSHOT");
-    send_key_until_needlematch("bootmenu-xen-kernel", 'down', 10, 5) if get_var('XEN');
-    if ((check_var('ARCH', 'aarch64') && is_sle && get_var('PLYMOUTH_DEBUG'))
-        || get_var('GRUB_KERNEL_OPTION_APPEND'))
-    {
-        $self->bug_workaround_bsc1005313 unless get_var("BOOT_TO_SNAPSHOT");
-    }
-    else {
-        # avoid timeout for booting to HDD
-        send_key 'ret';
-    }
+    grub_test();
```

Comment:
neatpicking, we could use some verb+name for this method, like 'handle_grub'

Reply:
I was also thinking about that name, but we already have `handle_grub` method name in opensusebasetest. So, I decided to use historically same name for the grub_test. Maybe, `test_grub` then?:)

Reply:
Then people would still wonder what's the difference to `handle_grub`. If that method already exists then just avoiding the name does not solve the ambiguity, right?

Reply:
I can only suggest 'process_grub' 'ensure_grub_present' if the other is taken, so it is also ok 'test_grub'

Reply:
There are not so many usages of `handle_grub`, so I we could have `handle_grub` for the method which handles more cases for booting and rename `handle_grub` in `opensusebasetest`. Actually it seems even that it has single usage in `sub wait_boot`, so we could even get rid of `handle_grub` by injecting it in a single place of usage.

Reply:
that sounds much better :)

Reply:
As far as I can see from the commits history, the method `grub_test` was extracted intentionally to improve readability of `wait_boot` method that uses the former one. Now as a result of the discussion I has to move it back to the `wait_boot`, that will revert back the work that was done before.

I see this issue as a little bit more than just naming conflict. From my point of view it is design issue, that requires some investigations and refactoring.

I can apply the solution, that is suggested in this discussion, if you insist. But this will remove work that was done before, though I believe it will be better to think on the solution as about separate problem and do not apply a makeshift (e.g. work in a separate PR that will address this particular problem).

Reply:
ok. Would you work on that before or after this PR? Either way is fine for me though

Reply:
We'll work on that after the PR. I'll create the appropriate ticket.